docs/adr/0003-deletion-strategy.md

```diff
@@ -0,0 +1,28 @@
+# 3. Deletion strategy
+
+* Status: proposed
+* Date: 2023-08-22
+* Authors: @furkatgofurov7
+* Deciders: @richardcase @alexander-demicev @salasberryfin @Danil-Grigorev @mjura
+
+## Context
+
+As operator, it is crucial to establish clear guidelines for how Rancher turtles handle cluster deletion operations. The objective is to facilitate the removal of a CAPI-imported cluster from Rancher, ensuring that it no longer appears in the Rancher UI, all while keeping the underlying CAPI cluster intact.
+
+## Decision
+
+The deletion workflow is structured as follows:
+
+**Ownership Chain:** Rancher cluster is associated with CAPI cluster through the owner references chain during their creation process.
+
+**Cluster Annotation:** When deleting a Rancher cluster, the operator will annotate the corresponding CAPI cluster with the `ClusterImportedAnnotation` annotation. This annotation will prevent automatic re-import of the CAPI cluster, only if the Rancher cluster itself contains the same annotation.
```

Comment:
From the name of the annotation it is not clear what is it actually set on the object. People should not have to look in the code after reading this ADR for an actual value. Might be better to also mention it is `imported` annotation here.